custom_components/toyota/sensor.py

```diff
@@ -318,7 +318,6 @@ def __init__(  # noqa: PLR0913
         super().__init__(coordinator, entry_id, vehicle_index, description)
         self.description = description
         self._attr_native_unit_of_measurement = unit
-        self._attr_suggested_unit_of_measurement = unit

```

Comment:
Are you shure? This will deactivate the suggested unit of measurement for all Sensors and not just for the fuel Level sensor.

Reply:
No, not fully sure.
The only information I have found regarding this is here: https://developers.home-assistant.io/docs/core/entity/sensor/
I am not really able to understand the difference between the native_unit_of_measurement and suggested_unit_of_measurement attributes.

Would you rather we just set the `suggested_unit_of_measurement` for `fuel_sensor` to `None`?

Reply:
> I am not really able to understand the difference between the native_unit_of_measurement and suggested_unit_of_measurement attributes.

I just tried to explain that in my comment 😉

Reply:
> Would you rather we just set the `suggested_unit_of_measurement` for `fuel_sensor` to `None`?

Yes. That would be a better approach.

Reply:
As discussed in #273, changing `unit_of_measurement` for fuel_sensor to `None` means we lose the unit completely.
My opinion is to to just remove the `suggested_unit_of_measurement` for the fuel_sensor.

I can think of multiple ways of solving this, but the most non-hacky approach seems to be to apply the suggest unit explicitly for all sensors.